### PR TITLE
Add action handler

### PR DESCRIPTION
**Pull request by edaubert**
Hi,

Here I propose a way to handle action with notify-send.

By defining an action you add a button on the notification which will call the corresponding command when the user clicks on it.

The implementation is quite simple :
- a daemon is launch to listen on dbus events which contains a specific string
- the action is passed on dbus notification message 
- The notification server sends a dbus message when the action is triggered by the user
- the message is received by the daemon which run the given command

Currently the command to run is set on the action itself so nothing is stored.
Only one daemon is run and each time notify-send is called, it checks if the notify-action daemon is running.

### CODE REVIEW

**Comment by vlevit** [comment]
@edaubert Hi! Sounds very cool! Thank you for sending this pull request. I'll try to review it very soon.

File: notify-action.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+NOTIFY_ARGS=(--session "path=/org/freedesktop/Notifications,member=ActionInvoked")
+
+# set -x
+
+dbus-monitor "${NOTIFY_ARGS[@]}" |
```
Comment (vlevit) [inline notify-action.sh]: Looks like `dbus-montitor` comes from `dbus` package while `gdbus` used in notify-send.sh is from `glib2` (`libglib2.0-bin` in Debian/Ubuntu). Let's mention `dbus-monitor` from `dbus` package dependency in README.md.

File: notify-action.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+NOTIFY_ARGS=(--session "path=/org/freedesktop/Notifications,member=ActionInvoked")
+
+# set -x
```
Comment (vlevit) [inline notify-action.sh]: debug leftover?

File: notify-send.sh [hunk 3]
```diff
@@ -19,6 +19,19 @@
 # Desktop Notifications Specification
 # https://developer.gnome.org/notification-spec/
 
+DIR=`dirname "$0"`
+
+# set -x
```
Comment (vlevit) [inline notify-send.sh]: debug leftover?

File: notify-send.sh [hunk 4]
```diff
@@ -44,6 +57,7 @@ Application Options:
   -i, --icon=ICON[,ICON...]         Specifies an icon filename or stock icon to display.
   -c, --category=TYPE[,TYPE...]     Specifies the notification category.
   -h, --hint=TYPE:NAME:VALUE        Specifies basic extra data to pass. Valid types are int, double, string and byte.
+  -o, --action=NAME:ACTION           Specifies action button which should be integrated to the notification. NAME will be dispayed, ACTION is the comman to run.
```
Comment (vlevit) [inline notify-send.sh]: Extra space before `Specifies...` (not aligned).

File: notify-send.sh [hunk 5]
```diff
@@ -61,6 +75,10 @@ convert_type() {
     esac
 }
 
+make_action() {
+  echo "\"notify-send: $2\", \"$1\""
```
Comment (vlevit) [inline notify-send.sh]: Is `notify-send` a uniquely chosen string by which we identify our actions? If so can it be `notify-send.sh` then?

File: notify-send.sh [hunk 6]
```diff
@@ -19,6 +19,19 @@
...
+# set -x
+
+run_action_handler () {
+  action=`ps axu | grep "notify-action.sh" | grep -v "grep"`
+  if [[ -z "$action" ]] ; then
+    echo "run_action_handler"
+    ($DIR/notify-action.sh 2> /dev/null &)
+  fi
+}
+run_action_handler
```
Comment (vlevit) [inline notify-send.sh]: Can we try to run action daemon only if `--action` is specified?

File: notify-send.sh [hunk 7]
```diff
@@ -19,6 +19,19 @@
...
 # https://developer.gnome.org/notification-spec/
 
+DIR=`dirname "$0"`
+
+# set -x
+
+run_action_handler () {
+  action=`ps axu | grep "notify-action.sh" | grep -v "grep"`
+  if [[ -z "$action" ]] ; then
+    echo "run_action_handler"
```
Comment (vlevit) [inline notify-send.sh]: Was this echo supposed to be left?

File: notify-send.sh (hunk 4 above)
Comment (vlevit) [inline notify-send.sh]: Specifying multiple actions is working by design, isn't it? I have tested and it worked for me.
Let's reflect this in command help and / or in example in README.md

Comment (edaubert) [reply notify-send.sh]: Well I reuse the same behavior that there is for the hints. I added

File: notify-action.sh (hunk 2 above)
Comment (edaubert) [reply notify-action.sh]: Yes I removed it

File: notify-action.sh (hunk 1 above)
Comment (edaubert) [reply notify-action.sh]: I added a word about it on the README.md

File: notify-send.sh (hunk 3 above)
Comment (edaubert) [reply notify-send.sh]: Yes I removed it

File: notify-send.sh (hunk 7 above)
Comment (edaubert) [reply notify-send.sh]: As of some `set -x`, this should have been removed. I did it

File: notify-send.sh (hunk 6 above)
Comment (edaubert) [reply notify-send.sh]: Done

File: notify-send.sh (hunk 4 above)
Comment (edaubert) [reply notify-send.sh]: Done

File: notify-send.sh (hunk 5 above)
Comment (edaubert) [reply notify-send.sh]: Done

**Comment by vlevit** [comment]
@edaubert Great contribution! Thank you.

**Pull request merged**